### PR TITLE
Redoing the project and creating the authorization

### PR DESCRIPTION
**Pull request by jrcfxx**

### CODE REVIEW

File: app/Http/Controllers/AuthController.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Validation\ValidationException;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Support\Facades\DB;
+
+class AuthController extends Controller
+{
+    public function __construct(
```
Comment (brunorconte) [inline app/Http/Controllers/AuthController.php:17]: missing comments

File: app/Http/Controllers/AuthController.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        private User $user,
+        private UsersController $userController
+    ) 
+    {
+        $this->userController = $userController;
+    }
+
+
+    public function register(Request $request)
```
Comment (brunorconte) [inline app/Http/Controllers/AuthController.php:26]: missing comments

File: app/Http/Controllers/AuthController.php [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    {
+        //Assigns the UsersController instance to the $userController property - available in other methods of the class
+        return $this->userController->create($request);
+    }
+
+    public function login(Request $request)
```
Comment (brunorconte) [inline app/Http/Controllers/AuthController.php:32]: missing comments

File: app/Http/Controllers/AuthController.php [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                    //dd($user->getAllPermissions($role)->pluck('name'));
+                    return response()->json(['token' => $token], 200);
+                }
+            } catch (\Exception $e) {
+                return response()->json(['message' => 'Internal Error'], 500);
+            }
+        }
+}
+
+    public function logout(Request $request)
```
Comment (brunorconte) [inline app/Http/Controllers/AuthController.php:75]: missing comments

File: app/Http/Controllers/PermissionController.php [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Permissions;
+use Illuminate\Http\Request;
+
+class PermissionController extends Controller
```
Comment (brunorconte) [inline app/Http/Controllers/PermissionController.php:8]: why is there an empty controller?

File: app/Http/Controllers/RoleController.php [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Role;
+use Illuminate\Http\Request;
+
+class RoleController extends Controller
```
Comment (brunorconte) [inline app/Http/Controllers/RoleController.php:8]: why is there an empty controller?

File: app/Http/Controllers/TaskChangeController.php [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+use App\Models\Task;
+use App\Models\TaskChange;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Support\Facades\DB;
+
+class TaskChangeController extends Controller
+{
+
+    public function __construct(private TaskChange $taskChange)
```
Comment (brunorconte) [inline app/Http/Controllers/TaskChangeController.php:14]: missing function description (params, return type, etc.)

File: app/Http/Controllers/TaskController.php [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+
+use App\Models\Task;
+use App\Models\TaskChange;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Support\Facades\DB;
+
+class TaskController extends Controller
+{
+    public function __construct(
```
Comment (brunorconte) [inline app/Http/Controllers/TaskController.php:13]: missing function description

File: app/Http/Controllers/TaskController.php [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        return Task::all();
+    }
+
+    /**
+     * Creating a new record in the database.
+     *
```
Comment (brunorconte) [inline app/Http/Controllers/TaskController.php:32]: missing param in the comments

File: app/Http/Controllers/UsersController.php [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+use App\Models\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+
+
+class UsersController extends Controller
+{
+    public function __construct(private User $user)
```
Comment (brunorconte) [inline app/Http/Controllers/UsersController.php:16]: missing comments about constructor description

File: app/Http/Controllers/UsersController.php [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+                DB::rollBack();
+                return response()->json(['message' => 'Internal Error'], 500);
+            }
+    }
+}
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  \App\Models\User  $user
```
Comment (brunorconte) [inline app/Http/Controllers/UsersController.php:78]: why is there a comment saying there is a param $user?

File: app/Http/Controllers/UsersController.php [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+            }
+        }
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \App\Models\User  $user
+     * @return \Illuminate\Http\Response
```
Comment (brunorconte) [inline app/Http/Controllers/UsersController.php:109]: recommend spacing @params and @return

Comment (brunorconte) [reply app/Http/Controllers/UsersController.php:109]: also wrong params here

File: app/Http/Controllers/UsersController.php [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+            return response()->json(['message' => 'Internal Error'], 500);
+
+        }
+    }
+    }
+
+    /**
+     * Remove the specified resource from storage.
+     *
+     * @param  \App\Models\User  $user
```
Comment (brunorconte) [inline app/Http/Controllers/UsersController.php:156]: wrong params

**Review by brunorconte — CHANGES_REQUESTED** [review]
Please revisit every Class, add and correct the constructor and methods description following this guideline:

/**
* This function does this and that
* 
* @param string $someParam
* @param string $someOtherParam
*
* @return void
*/

File: app/Http/Controllers/AuthController.php (hunk 3 above)
Comment (jrcfxx) [reply app/Http/Controllers/AuthController.php:59]: Done.

File: app/Http/Controllers/AuthController.php (hunk 4 above)
Comment (jrcfxx) [reply app/Http/Controllers/AuthController.php:110]: Done.

File: app/Http/Controllers/PermissionController.php (hunk 5 above)
Comment (jrcfxx) [reply app/Http/Controllers/PermissionController.php]: Done.

File: app/Http/Controllers/UsersController.php (hunk 11 above)
Comment (jrcfxx) [reply app/Http/Controllers/UsersController.php]: Fixed.

File: app/Http/Controllers/UsersController.php (hunk 13 above)
Comment (jrcfxx) [reply app/Http/Controllers/UsersController.php]: Fixed.

File: app/Http/Controllers/UsersController.php (hunk 12 above)
Comment (jrcfxx) [reply app/Http/Controllers/UsersController.php]: Fixed.

**Review by brunorconte — APPROVED** [review]

**Pull request merged**